client/asset/eth/multirpc.go

```diff
@@ -0,0 +1,806 @@
+}
+
+func (m *multiRPCClient) nonceProviderList() []*provider {
+	m.providerMtx.Lock()
+	defer m.providerMtx.Unlock()
+
+	providers := make([]*provider, 0, len(m.providers))
+
+	var lastProvider *provider
+	if time.Since(m.lastProvider.stamp) < nonceProviderStickiness {
+		lastProvider = m.lastProvider.provider
+	}
+
+	for _, p := range m.providers {
+		if lastProvider != nil && lastProvider.host == p.host {
+			continue // already added it
+		}
+		providers = append(providers, p)
+	}
+
+	rand.Shuffle(len(providers), func(i, j int) {
+		providers[i], providers[j] = providers[j], providers[i]
+	})
+
+	if lastProvider != nil {
+		providers = append([]*provider{lastProvider}, providers...)
+	}
+
+	return providers
+}
+
+func (m *multiRPCClient) nextNonce(ctx context.Context) (nonce uint64, err error) {
+	return nonce, m.withPreferred(func(p *provider) error {
+		nonce, err = p.cl().PendingNonceAt(ctx, m.creds.addr)
+		return err
+	})
+}
+
+func (m *multiRPCClient) address() common.Address {
+	return m.creds.addr
+}
+
+func (m *multiRPCClient) addressBalance(ctx context.Context, addr common.Address) (bal *big.Int, err error) {
+	return bal, m.withAny(func(p *provider) error {
+		bal, err = p.cl().BalanceAt(ctx, addr, nil /* latest */)
+		return err
+	})
+}
+
+func (m *multiRPCClient) bestHeader(ctx context.Context) (hdr *types.Header, err error) {
+	return hdr, m.withAny(func(p *provider) error {
+		hdr, err = p.bestHeader(ctx, m.log)
+		return err
+	})
+}
+
+func (m *multiRPCClient) chainConfig() *params.ChainConfig {
+	return m.cfg
+}
+
+func (m *multiRPCClient) peerCount() (c uint32) {
+	m.providerMtx.Lock()
+	defer m.providerMtx.Unlock()
+	for _, p := range m.providers {
+		if !p.failed() {
+			c++
+		}
+	}
+	return
+}
+
+func (m *multiRPCClient) contractBackend() bind.ContractBackend {
+	return m
+}
+
+func (m *multiRPCClient) lock() error {
+	return m.creds.ks.Lock(m.creds.addr)
+}
+
+func (m *multiRPCClient) locked() bool {
+	status, _ := m.creds.wallet.Status()
+	return status != "Unlocked"
+}
+
+func (m *multiRPCClient) pendingTransactions() ([]*types.Transaction, error) {
+	return []*types.Transaction{}, nil
+}
+
+func (m *multiRPCClient) shutdown() {
+	for _, p := range m.providerList() {
+		p.cl().Close()
+	}
+
+}
+
+func (m *multiRPCClient) sendSignedTransaction(ctx context.Context, tx *types.Transaction) error {
+	var lastProvider *provider
+	if err := m.withPreferred(func(p *provider) error {
+		lastProvider = p
+		m.log.Tracef("Sending signed tx via %q", p.host)
+		return p.cl().SendTransaction(ctx, tx)
+	}); err != nil {
+		return err
+	}
+	m.lastProvider.Lock()
+	m.lastProvider.provider = lastProvider
+	m.lastProvider.stamp = time.Now()
+	m.lastProvider.Unlock()
+	return nil
+}
+
+func (m *multiRPCClient) sendTransaction(ctx context.Context, txOpts *bind.TransactOpts, to common.Address, data []byte) (*types.Transaction, error) {
+	nonce, err := m.nextNonce(ctx)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This is based on code that's getting fixed in #1638 .  The nonce comes from txOps.Nonce, so shouldn't call `nextNonce` in here.
https://github.com/decred/dcrdex/pull/1638/files#diff-2603858a9b5db2a5039633341a2fd13fd7933ee616b3182cb83f32e6f3610cd3R228